### PR TITLE
[Refactor] `jsx-closing-bracket-location`, `jsx-no-bind`: fix eslint issues

### PR DESCRIPTION
**Pull request by caroline223**
@ljharb

### CODE REVIEW

**Review by ljharb — COMMENTED** [review]
I don't see any eslint overrides being removed, or config being changed?

**Comment by caroline223** [comment]
@ljharb May you review this PR for me? Thanks.

File: lib/rules/jsx-closing-bracket-location.js [hunk 1]
```diff
@@ -165,7 +165,7 @@ module.exports = {
      * @return {String} The characters used for indentation
      */
     function getIndentation(tokens, expectedLocation, correctColumn) {
-      correctColumn = correctColumn || 0;
+      const newColumn = correctColumn || 0;
```
Comment (caroline223) [inline lib/rules/jsx-closing-bracket-location.js:168]: @ljharb Here is where I made one change. Would this be plausible?

File: lib/rules/jsx-no-bind.js [hunk 2]
```diff
@@ -111,6 +114,11 @@ module.exports = {
       return null;
     }
 
+    /**
+     * @param {string | number} violationType
+     * @param {string} variableName
```
Comment (caroline223) [inline lib/rules/jsx-no-bind.js:119]: @ljharb May you review the changes that I have made here?

Comment (ljharb) [reply lib/rules/jsx-no-bind.js:119]: I think this PR needs to be rebased onto latest main, but in general it seems good :-)

**Comment by ljharb** [comment]
Synced with #3356.

**Pull request merged**